examples/azure/terraform-azure-automationaccount-example/main.tf

```diff
@@ -0,0 +1,197 @@
+# See test/azure/terraform_azure_automationaccount_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
+
+provider "azurerm" {
+  version = "~>2.20"
+  features {}
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# PIN TERRAFORM VERSION TO >= 0.12
+# The examples have been upgraded to 0.12 syntax
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.12"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A RESOURCE GROUP
+# See test/terraform_azure_nsg_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_resource_group" "automation_account_dsc_rg" {
+  name     = "${var.resource_group_name}-${var.postfix}"
+  location = var.location
+}
+
+resource "random_password" "virtual_machine" {
+  length           = 16
+  override_special = "-_%@"
+  min_upper        = "1"
+  min_lower        = "1"
+  min_numeric      = "1"
+  min_special      = "1"
+}
+
+resource "azurerm_automation_account" "automation_account" {
+  name                = "${var.automation_account_name}-${var.postfix}"
+  sku_name            = "Basic"
+  location            = azurerm_resource_group.automation_account_dsc_rg.location
+  resource_group_name = azurerm_resource_group.automation_account_dsc_rg.name
+}
+
+resource "azurerm_automation_dsc_configuration" "SampleDSC" {
+  name                    = var.sample_dsc_name
+  resource_group_name     = azurerm_resource_group.automation_account_dsc_rg.name
+  automation_account_name = azurerm_automation_account.automation_account.name
+  location                = azurerm_resource_group.automation_account_dsc_rg.location
+  content_embedded        = file(var.sample_dsc_path)
+  depends_on              = [azurerm_automation_account.automation_account]
+}
+
+# # ---------------------------------------------------------------------------------------------------------------------
+# # COMPILE THE SAMPLE DSC IN THE AUTOMATION ACCOUNT
+# # The Terraform `null_resource` signs in to Azure.  The second performs the compliation
+# # Compilation is triggered on every one to ensure the latest changes are always applied
+# # WARNING: ConvertTo-SecureString -String '${var.POWERSHELL_CLIENT_SECRET}, will expose the actual secret value in the logs.
+# # ---------------------------------------------------------------------------------------------------------------------
+resource "null_resource" "azureSignInPWSH" {
```

Comment:
Resource names should be underscore based (e.g., `azure_signing_pwsh`) for consistency with other examples. This applies to all resources in this example.